pkg/beacon/relay/entry/submission.go

```diff
@@ -77,7 +77,35 @@ func (res *relayEntrySubmitter) submitRelayEntry(
 					}
 					errorChannel <- err
 				})
-			return <-errorChannel
+
+			entryErr := <-errorChannel
```

Comment:
I think this will block until we can read from this channel and we don't always write to this channel. We should have a non-blocking read here with default return.

Reply:
We block here as before (`return <-errorChan`) and we always write either an error or `nil`. Actually I didn't changed the flow here.